src/persist/src/indexed/mod.rs

```diff
@@ -523,4 +548,60 @@ mod tests {
         i.seal(id, 3)?;
         Ok(())
     }
+
+    #[test]
+    fn buf_snapshot_validate() {
+        let range = SeqNo(1)..SeqNo(3);
+
+        // Normal case (equals lower)
+        let mut seqnos = BTreeSet::new();
+        seqnos.insert(SeqNo(1));
+
+        assert_eq!(
+            Indexed::<u64, u64, MemBuffer, MemBlob>::validate_buf_snapshot(&range, &seqnos),
```

Comment:
I couldn't think of a better way to call `validate_buf_snapshot` without these dummy type parameters (except for making it a free function which also seemed weird). I'm fine with this because its only clunky in tests, but open to all alternatives